### PR TITLE
Allow URL parameters to be passed in to `previous` and `next` links

### PR DESCRIPTION
**Pull request by curtisgibby**
Refs https://github.com/dereuromark/cakephp-calendar/issues/16;

### CODE REVIEW

File: src/Controller/Component/CalendarComponent.php [hunk 1]
```diff
@@ -64,10 +64,11 @@ public function startup(Event $event) {
 	 * @param string $year Year
 	 * @param string $month Month
 	 * @param int $span Years in both directions
+	 * @param array $options To be passed to previous and next methods
 	 * @throws \Cake\Http\Exception\NotFoundException
 	 * @return void
 	 */
-	public function init($year, $month, $span = 10) {
+	public function init($year, $month, $span = 10, $options = []) {
```
Comment (dereuromark) [inline src/Controller/Component/CalendarComponent.php:71]:
```suggestion
	public function init($year, $month, $span = 10, array $options = []) {
```

File: src/Controller/Component/CalendarComponent.php [hunk 2]
```diff
@@ -92,7 +93,16 @@ public function init($year, $month, $span = 10) {
 			throw new NotFoundException('Invalid date');
 		}
 
-		$this->Controller->set('_calendar', ['year' => $this->year, 'month' => $this->month, 'span' => $span]);
+		$viewVars = array_merge(
```
Comment (dereuromark) [inline src/Controller/Component/CalendarComponent.php:96]: For assoc arrays we should use `$new + $defaults`

File: src/View/Helper/CalendarHelper.php [hunk 3]
```diff
@@ -296,7 +296,17 @@ public function previousLink() {
...
 
-		return $this->Html->link(__('previous'), [$prevYear, $this->formatMonth($prevMonth)]);
+		$url = [
+			$prevYear,
+			$this->formatMonth($prevMonth),
+		];
+
+		$viewVars = $this->_View->get('_calendar');
+		if (!empty($viewVars['url'])) {
+			$url = array_merge($url, $viewVars['url']);
```
Comment (dereuromark) [inline src/View/Helper/CalendarHelper.php:306]: Why merging and not just replacing?
The merging sounds like it could go wrong in some cases.

Comment (curtisgibby) [reply src/View/Helper/CalendarHelper.php:306]: I don't want to require the userland code to set the year / month variables.

Comment (curtisgibby) [reply src/View/Helper/CalendarHelper.php:306]: > What if there are query strings on the current URL, how would the new URL query strings get merged in?

The existing implementation doesn't respect query strings from the current URL; that's what I'm trying to replicate.

File: src/Controller/Component/CalendarComponent.php (hunk 1 above)
Comment (curtisgibby) [reply src/Controller/Component/CalendarComponent.php]: Fixed in 177cdfceb6ae4b3047d913ab957e86ba11109a40

File: src/Controller/Component/CalendarComponent.php (hunk 2 above)
Comment (curtisgibby) [reply src/Controller/Component/CalendarComponent.php]: Fixed in d249cf98381db9a3127b7d7903d6a03082893cd0

**Comment by curtisgibby** [comment]
Thanks for merging and for tagging a release!

**Pull request merged**